projects/media-viewer/src/lib/annotations/comment-set/comment/text-highlight.directive.ts

```diff
@@ -0,0 +1,46 @@
+import { AfterViewChecked, Directive, ElementRef, Input } from '@angular/core';
+import uuid from 'uuid/v4';
+
+@Directive({
+  selector: '[mvTextHighlight]'
+})
+export class TextHighlightDirective implements AfterViewChecked {
+
+  @Input() textToHighlight: string;
+  private readonly uuid = uuid();
+
```

Comment:
why do u need to create a new uuid?

Reply:
just to make sure that I have a random id so that when I'm clearing the search highlights, I don't accidentally clear other spans